builder/softlayer/client.go

```diff
@@ -182,9 +183,41 @@ func (self SoftlayerClient) doHttpRequest(path string, requestType string, reque
 		return nil, err
 	}
 
+	if err := self.hasErrors(decodedResponse); err != nil {
+		return nil, err
+	}
+
 	return decodedResponse, nil
 }
 
+func (self SoftlayerClient) doHttpRequestMany(path string, requestType string, requestBody *bytes.Buffer) ([]interface{}, error) {
+	responseBody, err := self.doRawHttpRequest(path, requestType, requestBody)
+	if err != nil {
+		err := errors.New(fmt.Sprintf("Failed to get proper HTTP response from SoftLayer API"))
+		return nil, err
+	}
+
+	var decodedResponse interface{}
+	err = json.Unmarshal(responseBody, &decodedResponse)
+	if err != nil {
+		err := errors.New(fmt.Sprintf("Failed to decode JSON response from SoftLayer: %s | %s", responseBody, err))
+		return nil, err
+	}
+
+	switch v := decodedResponse.(type) {
+	case []interface{}:
+		return v, nil
+	case map[string]interface{}:
+		if err := self.hasErrors(v); err != nil {
+			return nil, err
+		}
+
+		return nil, errors.New("Expected multiple results")
+	default:
+		return nil, errors.New("Expected multiple results")
+	}
+}
+
```

Comment:
The return types of the functions differ: doHttpRequestMany returns an array. I don't know Go very well yet, but I think if I want to make a separate validation function that works with doHttpRequest, then I will have to change the return type of doHttpRequest from map[string]interface{} to interface{}. In that case, I will have to add a type cast in every existing location where doHttpRequest is called.

If you're okay with that, I'll make the change: it will be isolated within client.go. I won't need a separate validation function: doHttpRequest will return interface{}, and the act of casting it to an array or map[string]interface{} should suffice. I will probably still use the type switch so that it can call hasErrors().